packages/config/test/__fixtures__/has-e2e.config.mjs

```diff
@@ -0,0 +1,3 @@
+export default {
+  e2e: {}
+}
```

Comment:
Below file, `packages/config/test/ast-utils/addToCypressConfig.spec.ts` is marked as "too big to show diff" for some silly reason. It's the main part of this PR - the tests adding this missing coverage. Make sure you expand it and look.

The problem was I thought the `isusingEsModules` flag was the only thing I should check - this actually means "is using type: module". We need to include a check for `ts` which defaults to ESM.

We don't currently cover the case of using TS with CommonJS. I am not sure if this is a consideration we need to make, I've not seen any projects doing this.

Reply:
Here's the regression test described in the original issue: 

https://github.com/cypress-io/cypress/pull/21543/files#diff-8a7cf6dc3057090b692b41192285cc863933594137e517ae272da9b736a0196bR125-R155

I also added coverage for `mjs`.